src/modules/Lan.rb

```diff
@@ -1088,6 +997,68 @@ def activate_network_service
         LanItems.reload_config(ifaces)
       end
     end
+
+    def configure_as_bridge!(ifcfg, bridge_name)
+      return false if !NetworkInterfaces.Edit(ifcfg)
+
+      old_config = deep_copy(NetworkInterfaces.Current)
+      log.debug("Old Config #{ifcfg}\n#{old_config}")
+
+      log.info("old configuration #{ifcfg}, bridge #{bridge_name}")
+
+      NetworkInterfaces.Name = bridge_name
+
+      # from bridge interface remove all bonding-related stuff
```

Comment:
why only bonding stuff is being removed? Why not even vlan, ptp and so on? I think that using ```LanItems::SetDeviceMap``` would shorten this function a bit and it should remove direct NetworkInterfaces cache hacking.